public/app/plugins/datasource/loki/modifyQuery.ts

```diff
@@ -165,45 +166,45 @@ export function addLabelToQuery(
     return query;
   }
 
+  const parserPositions = getParserPositions(query);
+  const labelFilterPositions = getLabelFilterPositions(query);
   const hasStreamSelectorMatchers = getMatcherInStreamPositions(query);
   const everyStreamSelectorHasMatcher = streamSelectorPositions.every((streamSelectorPosition) =>
     hasStreamSelectorMatchers.some(
       (matcherPosition) =>
         matcherPosition.from >= streamSelectorPosition.from && matcherPosition.to <= streamSelectorPosition.to
     )
   );
-  const parserPositions = getParserPositions(query);
-  const labelFilterPositions = getLabelFilterPositions(query);
 
   const filter = toLabelFilter(key, value, operator);
   // If we have non-empty stream selector and parser/label filter, we want to add a new label filter after the last one.
   // If some of the stream selectors don't have matchers, we want to add new matcher to the all stream selectors.
```

Comment:
I think this comment should be removed as it is not relevant to case when we have `labelType`. And for the case when we don't have it, we have a comment there already.

Reply:
Removed in 66780e72c88314369e337c93c6712d8eba4ec782